build/dart/rules.gni

```diff
@@ -168,7 +169,129 @@ template("dart_snapshot") {
   }
 }
 
-template("dart_pkg_helper") {
+# Creates an app-jit snapshot for a command-line Dart program based on a
+# training run.
+#
+# Parameters:
+#  main_dart (required):
+#    The entrypoint to the Dart application.
+#
+#  training_args (required):
+#    Arguments to pass to the Dart application for the training run.
+#
+#  vm_args (optional):
+#    Additional arguments to the Dart VM.
+#
+#  deps (optional):
+#    Any build dependencies.
+#
+#  dot_packages (required):
+#    The .packages file for the app. Defaults to the $_dart_root/.packages.
+#
+#  output (optional):
+#    Overrides the full output path.
+#
+#  snapshot_kind (optional)
+#    Either an "app-jit" snapshot (default) or a "kernel" snapshot
+template("application_snapshot") {
+  assert(defined(invoker.main_dart), "Must specify 'main_dart'")
+  assert(defined(invoker.training_args), "Must specify 'training_args'")
+  assert(defined(invoker.dot_packages), "Must specify 'dot_packages'")
```

Comment:
nit: `package_config` right? We should probably be pointing to the json file since the .packages is deprecated.

Reply:
Good suggestion. Done.